### ISSUE TITLE
Metacat replication fails for datafiles with "&" in the filename

### ISSUE THREAD

**Issue by mbjones** [NCEAS/metacat#830]
---

Author Name: **ben leinfelder** (ben leinfelder)
Original Redmine Issue: 5710, https://projects.ecoinformatics.org/ecoinfo/issues/5710
Original Date: 2012-09-12
Original Assignee: ben leinfelder

---

When troubleshooting SANParks-SAEON, we encountered errors with the documentInfo parser on the target machine:

judithk.1128.1 has the filename: "KamathPL_ZebraSampling&GenotypeData.txt"
error:
knb 20120904-22:03:59: [ERROR]: ReplicationHandler.handleSingleDataFile - 6 Failed to write data file judithk.1128.1 into xml_documents from dataknp.sanparks.org/sanparks/servlet/replication because The reference to entity "GenotypeData.txt" must end with the ';' delimiter.

judithk.1223.1: "KNPEle & Buff Census 2007 Report.pdf"
error:
knb 20120906-04:10:23: [ERROR]: ReplicationHandler.handleSingleDataFile - 9 Failed to write data file judithk.1223.1 into xml_documents from dataknp.sanparks.org/sanparks/servlet/replication because The entity name must immediately follow the '&' in the entity reference. [ReplicationLogging]

**Comment by mbjones** [comment]
---

Original Redmine Comment
Author Name: **ben leinfelder** (ben leinfelder)
Original Date: 2012-09-12T21:05:20Z

---

I've fixed this and wrote a simple test that exercises the docInfo parser. I just wrapped the content of the 'docname' field in CDATA tags so that any reserved characters do not cause errors.
I'd like to include it in the 2.0.4 release, but we are in the midst of some somewhat-final testing. Action TBD.
